### DISCUSSION TITLE
KiTTY opens some text file hyperlinks in editor, others in browser

### DISCUSSION THREAD

**Discussion by SirWrexes** [kovidgoyal/kitty#5731] [Q&A] [answered]
Hi !

I am having a strange problem where, despite setting the action for MIME file protocol `text/*`, Kitty opens some text files in my editor (nvim) like it should, and some others in my web browser.

The trickiest part is that I don't really understand why, and it seems **very** arbitrary:
- If I do `echo 'some text' > some_file; hg 'some text'`  and click the link to the file, it opens in my browser.
- Some files like `JSON, md, js, ts` I know will work properly, or at least have worked properly in all my cases.
- However, `*.tsx` or `*.xml`, for example, open in the browser.

It seems to be dependant on the file extension somehow, although `xdg-mime query filetype` on each file return `text/plain`.

I am using Fish, on Arch Linux (Xanmod) and SwayWM (so Wayland), if it's any help.

Here are my `hg` function and my `open-actions.conf`:
```fish
function hg --wraps=rg
 kitty +kitten hyperlinked_grep $argv; 
end
```

```conf
# Open text files in favourites editor
protocol file
mime text/*
action launch --type=overlay "${EDITOR}" "${FILE_PATH}"

# Open images in a preview overlay
protocol file
mime image/*
action launch --type=overlay kitty +kitten icat --hold "${FILE_PATH}"

# Open executable file
protocol file
mime inode/executable,application/vnd.microsoft.portable-executable
action launch --hold --type=os-window "${FILE_PATH}"
```

**Comment by kovidgoyal** [comment, marked as answer]
some_file is not a text file. Use some_file.txt

And kitty does not use xdg-mime, it uses python's mimetypes module which
in turn uses /etc/mime.types

xml is application/xml
tsx is likely unknown

Remember that open actions can match by file extension as well as mime
type.

**Comment by SirWrexes** [comment]
Oh that makes things much clearer ! Thanks for your input.

After quickly checking out your docs for matching criteria, I have decided to make a `~/.config/kitty/mime.types` in which I wrote `text/source jsx tsx xml` and that I will add to if needed.

I love that it is an option !
